src/main/java/fallingleaves/fallingleaves/mixin/LeafTickMixin.java

```diff
@@ -78,14 +90,15 @@ public void randomDisplayTick(BlockState state, World world, BlockPos pos, Rando
                 float l = (float) (j >> 8 & 255) / 255.0F;
                 float m = (float) (j & 255) / 255.0F;
 
-                //Regular leaves
-                for (int leaf = 0; leaf < FallingLeaves.coniferLeaves.length; leaf++) {
-                    if (state.getBlock() == FallingLeaves.coniferLeaves[leaf]) {
-                        world.addParticle(FallingLeavesClient.FALLING_SPRUCE_LEAF, (double)pos.getX() + d, pos.getY(), (double)pos.getZ() + f, k, l, m);
-                    } else {
-                        world.addParticle(FallingLeavesClient.FALLING_LEAF, (double)pos.getX() + d, pos.getY(), (double)pos.getZ() + f, k, l, m);
+                boolean isConifer = false;
+                for (int i = 0; i < FallingLeavesClient.coniferLeaves.length; i++) {
+                    if (state.getBlock() == FallingLeavesClient.coniferLeaves[i]) {
+                        isConifer = true;
+                        break;
                     }
                 }
```

Comment:
This change would fix multiple leaves spawning if there is more than one entry in `coniferLeaves`